include/engine/engine.hpp

```diff
@@ -47,7 +48,7 @@ class Engine final
   public:
     Engine(EngineConfig &config_);
     Engine(const Engine &) = delete;
-    int RunQuery(const RouteParameters &route_parameters, util::json::Object &json_result);
+    Status RunQuery(const RouteParameters &route_parameters, util::json::Object &json_result);
```

Comment:
tl;dr: returning an incomplete type is UB

3.9/5 http://eel.is/c++draft/basic.types#5

> A class that has been declared but not defined, an enumeration type in certain contexts ([dcl.enum]), or an array of unknown size or of incomplete element type, is an incompletely-defined object type.45 Incompletely-defined object types and cv void are incomplete types ([basic.fundamental]).

7.2/6 http://eel.is/c++draft/dcl.enum#6

> An enumeration whose underlying type is fixed is an incomplete type from its point of declaration ([basic.scope.pdecl]) to immediately after its enum-base (if any), at which point it becomes a complete type. An enumeration whose underlying type is not fixed is an incomplete type from its point of declaration to immediately after the closing } of its enum-specifier, at which point it becomes a complete type.

3.10/4 http://eel.is/c++draft/basic.lval#4

> Unless otherwise indicated ([expr.call]), prvalues shall always have complete types or the void type; in addition to these types, glvalues can also have incomplete types. [ Note: class and array prvalues can have cv-qualified types; other prvalues always have cv-unqualified types. See Clause [expr].  — end note ]